mono/utils/mono-state.c

```diff
@@ -637,24 +647,21 @@ mono_crash_dump (const char *jsonFile, MonoStackHash *hashes)
 
 	// Save up to 100 dump files for a given stacktrace hash
 	for (int increment = 0; increment < 100; increment++) {
-		FILE* fp;
-		char *name = g_strdup_printf ("mono_crash.%" PRIx64 ".%d.json", hashes->offset_free_hash, increment);
-
-		if ((fp = fopen (name, "ab"))) {
-			if (ftell (fp) == 0) {
-				fwrite (jsonFile, size, 1, fp);
-				success = TRUE;
-			}
+		char name [100]; 
+		name [0] = '\0';
+		g_snprintf (name, sizeof (name), "mono_crash.%" PRIx64 ".%d.json", hashes->offset_free_hash, increment);
+
+		int handle = g_open (name, O_WRONLY | O_CREAT | O_EXCL);
+		if (handle == -1) {
```

Comment:
`g_assertf(…)` for the double fault.

Reply:
This is called from an icall. It seems a bit much to crash the runtime in that case.

Reply:
Also this isn't what we want because we are doing this loop to try to make file names that may already be taken.

Reply:
In practice, there are often multiple mono_crash files with the same stack hash